Doc/c-api/type.rst

```diff
@@ -217,6 +219,19 @@ The following functions and structs are used to create
       The function now accepts a single class as the *bases* argument and
       ``NULL`` as the ``tp_doc`` slot.
 
+.. c:function:: PyObject* PyType_FromMetaModuleAndSpec(PyTypeObject *meta, PyObject *module, PyType_Spec *spec, PyObject *bases)
+
+   Create and return a :ref:`heap type <heap-types>` from the *spec*
+   (:const:`Py_TPFLAGS_HEAPTYPE`). This function is a generalization of
+   :c:func:`PyType_FromModuleAndSpec`, with the main difference being that
+   the metaclass *meta* is used to construct the resulting type object.
+
+   This is useful for language binding tools, where it enables use cases such
+   as storing binding-related data structures within the type object or
+   intercepting type-related operations.
```

Comment:
I don't think that the Python documentation should list how functions are used, but be focused on the purpose of the functions and how they should be used.

Reply:
Ok, so you would like me to remove this sentence? (I can do that.)